elide-async/src/main/java/com/yahoo/elide/async/service/DefaultResultStorageEngine.java

```
@@ -0,0 +1,146 @@
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.AsyncQueryResultStorage;
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.request.EntityProjection;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.sql.Blob;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Iterator;
+
+import javax.inject.Singleton;
+import javax.sql.rowset.serial.SerialBlob;
+
+/**
+ * Utility class which implements ResultStorageEngine.
+ */
+@Singleton
+@Slf4j
+@Getter
+public class DefaultResultStorageEngine implements ResultStorageEngine {
+
+    @Setter private Elide elide;
+    @Setter private DataStore dataStore;
+    private String baseURL;
+
+    public DefaultResultStorageEngine() {
+    }
+
+    public DefaultResultStorageEngine(Elide elide, DataStore dataStore, String baseURL) {
+        this.elide = elide;
+        this.dataStore = dataStore;
+        this.baseURL = baseURL;
+    }
+
+    @Override
+    public URL storeResults(String asyncQueryID, byte[] byteResponse) {
+
+        AsyncQueryResultStorage aQRS = null;
+        try {
+            Blob response = new SerialBlob(byteResponse);
+
+            aQRS = (AsyncQueryResultStorage) DBUtil.executeInTransaction(elide, dataStore,
+                    (tx, scope) -> {
+
+                        AsyncQueryResultStorage asyncQueryResultStorage = new AsyncQueryResultStorage();
+                        asyncQueryResultStorage.setId(asyncQueryID);
+                        asyncQueryResultStorage.setResult(response);
+
+                        tx.save(asyncQueryResultStorage, scope);
+
+                        return asyncQueryResultStorage;
+                    });
+        } catch (SQLException e) {
+            log.error("Exception: {}", e);
+            throw new IllegalStateException(e);
+        } catch (Exception e) {
```

Let's not catch and swallow Exception (Too broad).